services/download/src/main/java/app/coronawarn/server/services/download/FederationBatchProcessor.java

```diff
@@ -25,7 +25,9 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Optional;
+import java.util.Queue;
 import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.commons.collections4.queue.CircularFifoQueue;
```

Comment:
in my opinion a bit over engineered... why isn't `java.util.Set` / HashSet sufficient?

Reply:
Removed the dependency and added a FIFO Queue with HashSet backing.